neuraxle/steps/flow.py

```diff
@@ -0,0 +1,217 @@
+            return self, data_container
+
+        self._nullify_hyperparams()
+        return self, data_container
+
+    def handle_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        """
+        Nullify wrapped step hyperparams, and don't transform the wrapped step.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :param context: execution context
+        :type context: ExecutionContext
+        :return: step, data_container
+        :type: DataContainer
+        """
+        if self.hyperparams[OPTIONAL_ENABLED_HYPERPARAM]:
+            return self.wrapped.handle_transform(data_container, context)
+
+        self._nullify_hyperparams()
+        return data_container
+
+    def _nullify_hyperparams(self):
+        """
+        Nullify wrapped step hyperparams using hyperparams_space.nullify().
+        """
+        hyperparams_space = self.wrapped.get_hyperparams_space()
+        self.wrapped.set_hyperparams(hyperparams_space.nullify())
+
+
+class ChooseOneOrManyStepsOf(Pipeline):
+    """
+    A pipeline to allow choosing many steps using an hyperparameter.
+
+    Example usage :
+
+    .. code-block:: python
+
+        p = Pipeline([
+            ChooseOneOrManyStepsOf([
+                ('a', Identity()),
+                ('b', Identity())
+            ])
+        ])
+        p.set_hyperparams({
+            'ChooseOneOrManyStepsOf__a__enabled': True,
+            'ChooseOneOrManyStepsOf__b__enabled': False
+        })
+        # or
+        p.set_hyperparams({
+            'ChooseOneOrManyStepsOf': {
+                'a': { 'enabled': True },
+                'b': { 'enabled': False }
+            }
+        })
+
+    .. seealso::
+        :class:`Pipeline`
+        :class:`Optional`
+    """
+
+    def __init__(self, steps, hyperparams=None):
+        Pipeline.__init__(self, steps)
+
+        if hyperparams is None:
+            self.set_hyperparams(HyperparameterSamples({}))
+        else:
+            self.set_hyperparams(hyperparams)
+
+        self._make_all_steps_optional()
+
+    def set_hyperparams(self, hyperparams: Union[HyperparameterSamples, OrderedDict, dict]) -> BaseStep:
+        """
+        Set hyperparams for step selection, and nullify the steps that are not chosen.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return: self
+        :rtype: BaseStep
+        """
+        super().set_hyperparams(hyperparams=hyperparams)
+
+        self._validate_choice_hyperparams()
+        self._set_all_hyperparams_steps_enabled_by_default()
+
+        return self
+
+    def _validate_choice_hyperparams(self):
+        for key in self.hyperparams.keys():
+            if key not in self.keys():
+                raise ValueError('Invalid Choosen Step {0} in {1}'.format(key, self.name))
```

Comment:
Is this method still useful? I don't think so. I think there is zero keys in hyperparams so this loops wont execute at all.